mysql-test/suite/innodb/r/fk_col_alter.result

```diff
@@ -132,3 +132,26 @@ ADD CONSTRAINT t2_fk FOREIGN KEY(f3) REFERENCES t1(f1);
 DROP TABLE t2, t1;
 SET SESSION FOREIGN_KEY_CHECKS = ON;
 # End of 10.4 tests
+#
+#  MDEV-34392 Inplace algorithm modifies the foreign key column
+#
+CREATE TABLE t1(f1 INT UNSIGNED PRIMARY KEY)ENGINE=InnoDB;
+CREATE TABLE t2(f1 INT UNSIGNED PRIMARY KEY,
+f2 INT UNSIGNED, INDEX(f2),
+CONSTRAINT fk_id FOREIGN KEY (f2) REFERENCES t1(f1)
+)ENGINE=InnoDB;
+SET FOREIGN_KEY_CHECKS=1;
+ALTER TABLE t2 MODIFY f2 INT UNSIGNED NOT NULL, ALGORITHM=COPY;
+ERROR HY000: Cannot change column 'f2': used in a foreign key constraint 'fk_id'
+ALTER TABLE t2 MODIFY f2 INT UNSIGNED NOT NULL, ALGORITHM=INPLACE;
+ERROR HY000: Column 'f2' cannot be NOT NULL: needed in a foreign key constraint 'test/fk_id' SET NULL
+SET FOREIGN_KEY_CHECKS=0;
+ALTER TABLE t2 MODIFY f2 INT UNSIGNED NOT NULL, ALGORITHM=COPY;
+ERROR HY000: Cannot change column 'f2': used in a foreign key constraint 'fk_id'
+ALTER TABLE t2 MODIFY f2 INT UNSIGNED NOT NULL, ALGORITHM=INPLACE;
+ERROR HY000: Column 'f2' cannot be NOT NULL: needed in a foreign key constraint 'test/fk_id' SET NULL
```

Comment:
There is no `ON DELETE SET NULL` or `ON UPDATE SET NULL` clause in `CONSTRAINT fk_id FOREIGN KEY (f2) REFERENCES t1(f1)`.